subprojects/jacoco/src/main/java/org/gradle/testing/jacoco/plugins/JacocoPlugin.java

```diff
@@ -109,18 +109,16 @@ private void configureCoverageDataElementsVariants(Project project) {
 
             testSuites.withType(JvmTestSuite.class).configureEach(suite -> {
                 suite.getTargets().configureEach(target -> {
-                    createCoverageDataVariant(project, suite, target);
+                    createCoverageDataVariant((ProjectInternal) project, suite, target);
                 });
             });
         });
     }
 
-    private void createCoverageDataVariant(Project project, JvmTestSuite suite, JvmTestSuiteTarget target) {
-        final Configuration variant = project.getConfigurations().create(COVERAGE_DATA_ELEMENTS_VARIANT_PREFIX + StringUtils.capitalize(target.getName()));
+    private void createCoverageDataVariant(ProjectInternal project, JvmTestSuite suite, JvmTestSuiteTarget target) {
+        final Configuration variant = project.getConfigurations().consumable(COVERAGE_DATA_ELEMENTS_VARIANT_PREFIX + StringUtils.capitalize(target.getName()));
```

Comment:
I think this needs to be a consumable bucket. We might need to use a custom role here to define a consumable/bucket which is deprecated for declaration.